README.md

```diff
@@ -472,6 +487,163 @@ async function saveCoverageData (codeBlocks) {
+import fs from 'fs'
+import saveToFile from './saveToFile'
+import standard from 'standard'
+import forEachCodeBlock from './forEachCodeBlock'
+
+export const paddingText = (width, string, padding = ' ') => {
+  return (width <= string.length) ? string : paddingText(width, string + padding, padding)
+}
+
+export const countLinterErrors = (results) => results[0].messages.length
+
+export const runStandardLinter = (contents, fix) => (
+  new Promise((resolve, reject) => {
+    standard.lintText(contents, { fix }, (err, { results }) => {
+      if (err) reject(err)
+      else resolve(results)
+    })
+  })
+)
+
+export const formatLineLinterError = (error) => (
+  paddingText(10, error.line + ':' + error.column + ': ') +
+  paddingText(20, error.filename) +
+  error.message + ' ' + '(' + error.ruleId + ')'
+)
+
+export const formatLinterError = (line, filename, output) => (error) => {
+  error.line += line - 1
+  error.filename = filename
+  error.fixedCode = output
+  return error
+}
+
+export const printLinterErrors = (errors) => {
+  console.error(errors.map(formatLineLinterError).join('\n'))
+}
+
+export const isFix = (options) => !!options._ && options._[0] === 'fix'
+
+export const insertCodeBlock = (code, filename) => {
+  const END = '`' + '`' + '`'
+  const BEGIN = END + 'js'
+  return [
+    BEGIN,
+    '// ' + filename,
+    code,
+    END
+  ].join('\n')
+}
+
+export const replaceAll = (text, pattern, replace) => text.split(pattern).join(replace)
+
+export const fixLinterErrors = async (errors, codeBlocks, targetPath = 'README.md') => {
+  let readme = fs.readFileSync(targetPath, 'utf8')
+  errors.map(({ filename, fixedCode }) => {
+    const code = codeBlocks[filename].contents.replace(/\n$/g, '')
+    if (fixedCode) {
+      readme = replaceAll(readme, insertCodeBlock(code, filename), insertCodeBlock(fixedCode, filename))
+    }
+  })
+  await saveToFile(targetPath, readme)
+}
+
+export async function runLinter (codeBlocks, options) {
+  let errors = []
+  const fix = isFix(options)
+  await forEachCodeBlock(async ({ contents }, filename, codeBlocks) => {
+    const { line } = codeBlocks[filename]
+    const results = await runStandardLinter(contents, fix)
+    results.map(({ messages, output }) => {
+      const newErrors = messages.map(formatLinterError(line, filename, output))
+      errors = errors.concat(newErrors)
+    })
+  })(codeBlocks)
+  if (fix) await fixLinterErrors(errors, codeBlocks)
+  else printLinterErrors(errors)
+}
+
+export default runLinter
+```
+
+And its tests
+
+```js
+// runLinter.test.js
+import {
+  paddingText,
+  runStandardLinter,
+  formatLineLinterError,
+  formatLinterError,
+  countLinterErrors,
+  isFix,
+  insertCodeBlock,
+  replaceAll
+} from './runLinter'
+
+it('should create padding text', () => {
+  assert(paddingText(5, 'hello') === 'hello')
+  assert(paddingText(6, 'hello') === 'hello ')
+  assert(paddingText(7, 'hello', '_') === 'hello__')
+})
+
+it('should format linter error', () => {
```

Comment:
Some methods are simply utility method and is part of internal implementation. I think you should reduce the test case to these:

- should map linter errors back to line in README.md
- should format linter errors on a column mode
- should trigger fix mode when `--fix` option is given

The utility functions such as `replaceAll` and `generateCodeBlock` should be covered by other tests. Then you don’t need to export them.